buildRustPackage: remove git from nativeBuildInputs

Atemu opened the pull request:
Having git in nativeBuildInputs causes thousands of rebuilds when git is touched because every derivation that somehow transiently depends on a rust package will change. See https://github.com/NixOS/nixpkgs/pull/205682 for instance.

AFAICT git is unused. Only the fetcher needs git which it has already.

Fixes https://github.com/NixOS/nixpkgs/issues/205804

###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

Atemu committed:
buildRustPackage: remove git from nativeBuildInputs

Having git in nativeBuildInputs causes thousands of rebuilds when git is touched
because every derivation that somehow transiently depends on a rust package will
change. See https://github.com/NixOS/nixpkgs/pull/205682 for instance.

AFAICT git is unused. Only the fetcher needs git which it has already.

Fixes https://github.com/NixOS/nixpkgs/issues/205804

figsoda commented on pkgs/build-support/rust/build-rust-package/default.nix:
```diff
@@ -117,7 +117,6 @@ stdenv.mkDerivation ((removeAttrs args [ "depsExtraArgs" "cargoUpdateHook" "carg
 
   nativeBuildInputs = nativeBuildInputs ++ [
     cacert
```
iiuc cacert can be removed here as well?

Atemu replied:
Good point, cacert shouldn't even have any "native" parts and would at most be in `buildInputs`. 

This has more potential for breakage though as I can imagine some builds might have depended on cacert before (without declaring it).  
If a build depends on something it doesn't declare, that's a bug though and it should be fixed by its maintainers. I think we can apply the pull principle here.

Atemu replied:
Do you know any rust packages that do internet stuff and might depend on cacert implicitly?

Atemu replied:
I've moved it to buildInputs. I can't gauge whether removing it is a good idea or not. If you think it can, please go ahead and remove it in another PR.

Atemu committed:
buildRustPackage: move cacert to buildInputs

Cacert doesn't have any binary parts; it does not make sense to have it in
nativeBuildInputs. Its home sould be buildInputs.

It probably shouldn't be in there either as I can't imagine it being a general
requirement for rust packages and any package that depends on cacert should do
so explicitly. I don't have the experience with the rust packages set to know
whether that's the case, so let's err on the side of caution.

danieldk committed:
python3Packages.wandb: fix build

The logging test of `wandb` failed, because another package
(`PythonGit`) clutters the logs about not being able to confirm
whether we are running in a Cygwin environment. So, simply
disable the test.

raphaelr commented:
The `git` in the function parameters can be removed as well.

Atemu commented:
Indeed. The previous version I was looking at still passed git through to the fetcher.

dotlambda committed:
rapidfuzz-cpp: 1.10.1 -> 1.10.4

https://github.com/maxbachmann/rapidfuzz-cpp/blob/v1.10.4/CHANGELOG.md

fabaff committed:
python310Packages.aioshelly: 5.1.1 -> 5.2.0

Changelog: https://github.com/home-assistant-libs/aioshelly/releases/tag/5.2.0

samuela commented:
looks like there's an unrelated change to wandb in this PR as well. i'd recommend rebasing now that https://github.com/NixOS/nixpkgs/pull/205898 has landed

Atemu commented:
I think what happened is that I rebased on top of master thinking that would resolve the merge conflict but the conflicting change was actually on staging, so I rebased on top of that. However, staging didn't have all of master's commits merged in yet, so those other commits from master were also rebased on top of staging. Urgh.

Continued in https://github.com/NixOS/nixpkgs/pull/206195
